### ISSUE TITLE
ios build fails with cloud_firestore plugin

### ISSUE THREAD

**Issue by authentic-network** [flutter/flutter#32396]
Steps to reproduce:

- flutter create test
- add "cloud_firestore: 0.11.0+1" to pubspec.yaml
- flutter build ios breaks with:

/Users/xy/dev/an/an_test/ios/Pods/FirebaseDatabase/Firebase/Database/third_party/Wrap-leveldb/APLevelDB.mm:54:9: fatal error:
    'leveldb/db.h' file not found
    #import "leveldb/db.h"
            ^~~~~~~~~~~~~~
    1 error generated.

flutter doctor -v
[✓] Flutter (Channel unknown, v1.5.8, on Mac OS X 10.14.4 18E227, locale de-DE)
    • Flutter version 1.5.8 at /Users/xy/dev/flutter/flutter
    • Framework revision 0ba67226ee (2 weeks ago), 2019-04-24 17:18:28 -0700
    • Engine revision c63d1cf9c9
    • Dart version 2.3.0 (build 2.3.0-dev.0.1 1f1592edce)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/tstein/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2.1, Build version 10E1001
    • ios-deploy 1.9.4
    • CocoaPods version 1.7.0.rc.1

[!] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] Connected device (3 available)
....

name: an_test
description: A new Flutter project.
version: 1.0.0+1

environment:
  sdk: ">=2.1.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^0.1.2
  cloud_firestore: 0.11.0+1

dev_dependencies:
  flutter_test:
    sdk: flutter

# The following section is specific to Flutter.
flutter:
  uses-material-design: true

Podfile.lock:

PODS:
  - BoringSSL-GRPC (0.0.3):
    - BoringSSL-GRPC/Implementation (= 0.0.3)
    - BoringSSL-GRPC/Interface (= 0.0.3)
  - BoringSSL-GRPC/Implementation (0.0.3):
    - BoringSSL-GRPC/Interface (= 0.0.3)
  - BoringSSL-GRPC/Interface (0.0.3)
  - cloud_firestore (0.0.1):
    - Firebase/Auth
    - Firebase/Core
    - Firebase/Database
    - Firebase/Firestore (~> 6.0)
    - Flutter
  - Firebase/Auth (6.0.0):
    - Firebase/CoreOnly
    - FirebaseAuth (~> 6.0.0)
  - Firebase/Core (6.0.0):
    - Firebase/CoreOnly
    - FirebaseAnalytics (= 6.0.0)
  - Firebase/CoreOnly (6.0.0):
    - FirebaseCore (= 6.0.0)
  - Firebase/Database (6.0.0):
    - Firebase/CoreOnly
    - FirebaseDatabase (~> 6.0.0)
  - Firebase/Firestore (6.0.0):
    - Firebase/CoreOnly
    - FirebaseFirestore (~> 1.3.0)
  - firebase_core (0.0.1):
    - Firebase/Core
    - Flutter
  - FirebaseAnalytics (6.0.0):
    - FirebaseCore (~> 6.0)
    - FirebaseInstanceID (~> 4.0)
    - GoogleAppMeasurement (= 6.0.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 6.0)
    - GoogleUtilities/MethodSwizzler (~> 6.0)
    - GoogleUtilities/Network (~> 6.0)
    - "GoogleUtilities/NSData+zlib (~> 6.0)"
    - nanopb (~> 0.3)
  - FirebaseAuth (6.0.0):
    - FirebaseAuthInterop (~> 1.0)
    - FirebaseCore (~> 6.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 6.0)
    - GoogleUtilities/Environment (~> 6.0)
    - GTMSessionFetcher/Core (~> 1.1)
  - FirebaseAuthInterop (1.0.0)
  - FirebaseCore (6.0.0):
    - GoogleUtilities/Environment (~> 6.0)
    - GoogleUtilities/Logger (~> 6.0)
  - FirebaseDatabase (6.0.0):
    - FirebaseAuthInterop (~> 1.0)
    - FirebaseCore (~> 6.0)
    - leveldb-library (~> 1.18)
  - FirebaseFirestore (1.3.0):
    - FirebaseAuthInterop (~> 1.0)
    - FirebaseCore (~> 6.0)
    - FirebaseFirestore/abseil-cpp (= 1.3.0)
    - "gRPC-C++ (= 0.0.8)"
    - leveldb-library (~> 1.20)
    - nanopb (~> 0.3.901)
    - Protobuf (~> 3.1)
  - FirebaseFirestore/abseil-cpp (1.3.0):
    - FirebaseAuthInterop (~> 1.0)
    - FirebaseCore (~> 6.0)
    - "gRPC-C++ (= 0.0.8)"
    - leveldb-library (~> 1.20)
    - nanopb (~> 0.3.901)
    - Protobuf (~> 3.1)
  - FirebaseInstanceID (4.0.0):
    - FirebaseCore (~> 6.0)
    - GoogleUtilities/Environment (~> 6.0)
    - GoogleUtilities/UserDefaults (~> 6.0)
  - Flutter (1.0.0)
  - GoogleAppMeasurement (6.0.0):
    - GoogleUtilities/AppDelegateSwizzler (~> 6.0)
    - GoogleUtilities/MethodSwizzler (~> 6.0)
    - GoogleUtilities/Network (~> 6.0)
    - "GoogleUtilities/NSData+zlib (~> 6.0)"
    - nanopb (~> 0.3)
  - GoogleUtilities/AppDelegateSwizzler (6.0.0):
    - GoogleUtilities/Environment
    - GoogleUtilities/Logger
    - GoogleUtilities/Network
  - GoogleUtilities/Environment (6.0.0)
  - GoogleUtilities/Logger (6.0.0):
    - GoogleUtilities/Environment
  - GoogleUtilities/MethodSwizzler (6.0.0):
    - GoogleUtilities/Logger
  - GoogleUtilities/Network (6.0.0):
    - GoogleUtilities/Logger
    - "GoogleUtilities/NSData+zlib"
    - GoogleUtilities/Reachability
  - "GoogleUtilities/NSData+zlib (6.0.0)"
  - GoogleUtilities/Reachability (6.0.0):
    - GoogleUtilities/Logger
  - GoogleUtilities/UserDefaults (6.0.0):
    - GoogleUtilities/Logger
  - "gRPC-C++ (0.0.8)":
    - "gRPC-C++/Implementation (= 0.0.8)"
    - "gRPC-C++/Interface (= 0.0.8)"
  - "gRPC-C++/Implementation (0.0.8)":
    - "gRPC-C++/Interface (= 0.0.8)"
    - gRPC-Core (= 1.19.0)
    - nanopb (~> 0.3)
  - "gRPC-C++/Interface (0.0.8)"
  - gRPC-Core (1.19.0):
    - gRPC-Core/Implementation (= 1.19.0)
    - gRPC-Core/Interface (= 1.19.0)
  - gRPC-Core/Implementation (1.19.0):
    - BoringSSL-GRPC (= 0.0.3)
    - gRPC-Core/Interface (= 1.19.0)
    - nanopb (~> 0.3)
  - gRPC-Core/Interface (1.19.0)
  - GTMSessionFetcher/Core (1.2.1)
  - leveldb-library (1.20)
  - nanopb (0.3.901):
    - nanopb/decode (= 0.3.901)
    - nanopb/encode (= 0.3.901)
  - nanopb/decode (0.3.901)
  - nanopb/encode (0.3.901)
  - Protobuf (3.7.0)

DEPENDENCIES:
  - cloud_firestore (from `.symlinks/plugins/cloud_firestore/ios`)
  - firebase_core (from `.symlinks/plugins/firebase_core/ios`)
  - Flutter (from `.symlinks/flutter/ios-release`)

SPEC REPOS:
  https://github.com/cocoapods/specs.git:
    - BoringSSL-GRPC
    - Firebase
    - FirebaseAnalytics
    - FirebaseAuth
    - FirebaseAuthInterop
    - FirebaseCore
    - FirebaseDatabase
    - FirebaseFirestore
    - FirebaseInstanceID
    - GoogleAppMeasurement
    - GoogleUtilities
    - "gRPC-C++"
    - gRPC-Core
    - GTMSessionFetcher
    - leveldb-library
    - nanopb
    - Protobuf

EXTERNAL SOURCES:
  cloud_firestore:
    :path: ".symlinks/plugins/cloud_firestore/ios"
  firebase_core:
    :path: ".symlinks/plugins/firebase_core/ios"
  Flutter:
    :path: ".symlinks/flutter/ios-release"

SPEC CHECKSUMS:
  BoringSSL-GRPC: db8764df3204ccea016e1c8dd15d9a9ad63ff318
  cloud_firestore: ab9e897ba5cba56f0a722baa2a659204703f5756
  Firebase: fa80b9d987ca014a1ba9357496ef2a0178b28b12
  firebase_core: ce5006bb48508ee4e71e0f429a3f519bb8ee2961
  FirebaseAnalytics: 1743c5f4de3687d0745709dfdc4b1dea1484f44c
  FirebaseAuth: efc72b0f6fe075d2a6de8d81a6d9de83ef2b47eb
  FirebaseAuthInterop: 0ffa57668be100582bb7643d4fcb7615496c41fc
  FirebaseCore: e38f025287b413255a53acc1945d048a112047f7
  FirebaseDatabase: f48e067716864be2b855cf716b927ef375d6cfa0
  FirebaseFirestore: 0537b2a960a4819fae2834e688e5daeb571c81d6
  FirebaseInstanceID: 0e0348a3c00a734fa376a070f5ad4533ad975cb5
  Flutter: 58dd7d1b27887414a370fcccb9e645c08ffd7a6a
  GoogleAppMeasurement: 7f028ea162b72c8f326daec74afc95d94f7a47d6
  GoogleUtilities: f1faafc033ea203adf1783ce00af455bb99d0e5b
  "gRPC-C++": 98be881723177d8c4faf5fdffacb582c7b4971a2
  gRPC-Core: bd9472c8daa2e414b9f8038ba667bf56ce0e02b8
  GTMSessionFetcher: 32aeca0aa144acea523e1c8e053089dec2cb98ca
  leveldb-library: 08cba283675b7ed2d99629a4bc5fd052cd2bb6a5
  nanopb: 2901f78ea1b7b4015c860c2fdd1ea2fee1a18d48
  Protobuf: 7a877b7f3e5964e3fce995e2eb323dbc6831bb5a

PODFILE CHECKSUM: aff02bfeed411c636180d6812254b2daeea14d09

COCOAPODS: 1.7.0.rc.1
`

**Comment by logemann** [comment]
same here.

**Comment by praneybehl** [comment]
Same here 😞

**Comment by collinjackson** [comment]
@praneybehl the multidex error seems like it might be an unrelated issue, could you open a separate issue and also note if you've followed the steps at https://developer.android.com/tools/building/multidex.html

**Comment by martin-schulze-vireso** [comment]
Might be this: https://github.com/CocoaPods/CocoaPods/issues/8765

**Comment by collinjackson** [comment]
I think that's a good theory @martin-schulze-vireso. @authentic-network does switching to a different CocoaPods version fix the issue for you?

**Comment by praneybehl** [comment]
Thanks @collinjackson @martin-schulze-vireso, down grading cocoaPods fixed the issue with cloud_firestore.
 
I was originally at cocoapods-1.7.0.rc.1 now downgrading to v 1.7.0.beta.3 fixed it.

@authentic-network @logemann if you are still looking for a fix try doing the following:

1. to remove your current version you could just run:
`sudo gem uninstall cocoapods`
select the version : "cocoapods-1.7.0.rc.1"
2. install v 1.7.0.beta.3:
`sudo gem install cocoapods -v 1.7.0.beta.3`
3. delete the ./build folder, ./ios/Pods and Podfile.lock
4. select your ios device and run 👍 

Hope it helps, wasted so many hours on fixing it.

Cheers

**Comment by authentic-network** [comment]
I checked the suggested solution. It works. Thanks.

**Comment by sofierPoS** [comment]
any tried cocoapods-1.7.0.rc.2 and see if it fixes the issue=

**Comment by praneybehl** [comment]
@sofierPoS, why don’t you try and share?

**Comment by sofierPoS** [comment]
So far I can say I tried your receipt and didn't have to do step 3 to make it work, so it was pretty painless. Then I had to move on with a delivery that was blocked by this.
But I will try to install rc.2 and see how that goes.
Thanks.

**Comment by sofierPoS** [comment]
Ok, I tried it:
``
sudo gem install cocoapods --pre
``

I am now on 1.7.0.rc.2 (pod --version) and the build went through without issues.

**Comment by kroikie** [comment]
@authentic-network

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/375. Any further collaboration will be done there.
